cli/gumball-machine/README.md

```diff
@@ -0,0 +1,184 @@
+# Gumball Machine CLI
+
+## Getting Started
+
+See README.md in the parent directory.
+
+## Usage
+
+This is a CLI for Gumball Machine which generally sits on top of the TypeScript SDK in `candyland/contracts/sdk/gumball-machine`. It provides CLI commands for each core instruction that Gumball Machine supports. Note that before using the CLI, it's critical that you have access to an RPC URL for a network which has the entire Candyland suite deployed (SugarShack not required).
+
+### Key Design Choice
+
+In general, each command corresponds to an instruction in the Gumball Machine smart contract. All accounts that an instruction requires are generally specified as required options for the corresponding command. For more information about the options/arguements of each command you can run `ts-node gumball-machine-cli.ts <command_name> help`. 
+
+However, in general, the *arguements* to each instruction are specified in the template file associated with each command. See `/templates` for more. To be clear, a user should modify these template files each time they wish to change the arguements they are passing to the associated on-chain instruction.
+
+The reason for this choice is to best leverage the auto-generated Gumball-Machine Solita SDK. For example, rather than the user specifying the arguements to `init` via a JSON file (which would require its own spec), they fill in the same fields into the Solita TS type associated with `initialize_gumball_machine` on-chain. This is very helpful, especially since the contract is in semi-active development -- when the Solita SDK changes, the CLI will fail loudly, and can be easily updated.
+
+The drawback here is that some parameters are given at the CLI, and some in TS files, which could be a messy dev experience. However, care has been taken to make the TS templates as easy to use as possible. And specifying all of the arguements to, for instance, `init` at the CLI would be a clunky dev experience.
+
+## Docs
+
+### init
+
+Corresponds to `initialize_gumball_machine`. Creates a `GumballMachine` account and the underlying `MerkleRoll` account, paid by `payer`. The properties for the `GumballMachineHeader` associated with the account are set in: `templates/initializeGumballMachine`. 
+
+Example usage: Creating a GumballMachine taking payment in SOL off of localnet.
+
+`ts-node gumball-machine-cli.ts init -u "http://127.0.0.1:8899" -c "/Users/samorend/Documents/candyland/cli/gumball-machine/creator-address.json" -m So11111111111111111111111111111111111111112`
+
+Note: -u specifies the RPC url to be used, -c specifies the path to a keypair which will be the creator of the gumball machine (must sign off on subsequent updates), -m specifies the pubkey of the mint to be used for payments.
+
+Example `templates/initializeGumballMachine.ts`:
```

Comment:
Can you choose a different name for `templates`? These are not `templates` since they are actually complete instruction data sets

Reply:
technically the DispenseNFTForSol is actually template, bc most of the time I will probably want to default to dispensing a single NFT. 

Maybe restructure?
```
templates/
    actual default arguments for each instruction
arguments/
    *ts files that combined user args with template args & checks against generated Solita arg type
```

Reply:
agree, `templates` should be renamed. Right now, `templates` is basically just `arguments`.